fish-rust/src/wutil/mod.rs

```diff
@@ -68,13 +68,13 @@ pub fn wperror(s: &wstr) {
 
 /// Port of the wide-string wperror from `src/wutil.cpp` but for rust `&str`.
 pub fn perror(s: &str) {
-    let e = errno().0;
+    let e = Errno::last();
```

Comment:
The weird thing about `nix::errno::Errno` is that they treat "unknown error" the same as "no error".
This could cause subtle issues if the OS ever returns an error that hasn't been added to `nix` yet.
Unless they have a way of guaranteeing consistency with the host libc?

---

I'm not sure `enum` is the right type for `Errno`.
It's already marked `#[non_exhaustive]` so client code can't do exhaustive matching.
I guess they could make it:

```
enum Errno {
    NoError = 0
    EPERM = libc::EPERM,
    ...
    UnknownErrno(i32),
}
```

but at that point it's probably better to use a struct that wraps `libc::c_int` (or `i32`?), similar to `std::io::Error`.

---

I guess we could use the `nix::errno::Errno::last_raw()` (which is `nix::errno::errno()` today) and `nix::errno::Errno::set_raw()` precisely in the cases where we don't want to ignore unknown errors.
But the problem of being easy to misuse remains.